core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -515,8 +515,34 @@ private CompiledQuery alterTable(SqlExecutionContext executionContext) throws Sq
                     } else {
                         throw SqlException.$(lexer.lastTokenPosition(), "'param' expected");
                     }
+                } else if (SqlKeywords.isResumeKeyword(tok)) {
+                    tok = expectToken(lexer, "'wal'");
+                    if (!SqlKeywords.isWalKeyword(tok)) {
+                        throw SqlException.$(lexer.lastTokenPosition(), "'wal' expected");
+                    }
+
+                    tok = SqlUtil.fetchNext(lexer); // optional from part
+                    long fromTxn = -1;
+                    if (tok != null) {
+                        if (SqlKeywords.isFromKeyword(tok)) {
+                            tok = expectToken(lexer, "'transaction' or 'txn'");
+                            if (!(SqlKeywords.isTransactionKeyword(tok) || SqlKeywords.isTxnKeyword(tok))) {
+                                throw SqlException.$(lexer.lastTokenPosition(), "'transaction' or 'txn' expected");
+                            }
+                            CharSequence txnValue = expectToken(lexer, "transaction value");
+                            final int valuePosition = lexer.getPosition();
```

Comment:
this should be `last token position` otherwise error position will be incorrect for the value

Reply:
Fixed in https://github.com/questdb/questdb/pull/2841/commits/1245bf8516578159fb09b23a263ae9322113cf18